src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorage.cs

```diff
@@ -55,8 +56,9 @@ public void Init()
             }
         }
 
-        public IEnumerable<string> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
+        public Task<IEnumerable<string>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
         {
+            var subscribers = new List<string>();
             var result = new List<string>();
             var messagelist = messageTypes.ToList();
             lock (locker)
```

Comment:
@danielmarbach dumb question, but wouldn't this destroy async and block? Should we resort to an ["async lock"](http://blogs.msdn.com/b/pfxteam/archive/2012/02/12/10266988.aspx)?